pycfmodel/model/resources/resource.py

```diff
@@ -43,4 +45,28 @@ def has_hardcoded_credentials(self) -> bool:
 
     @property
     def policy_documents(self) -> List[OptionallyNamedPolicyDocument]:
-        return []
+        """
+        Returns a list with all the optionally named policy documents in this resource within its properties.
+        Every resource has a Properties field, if not, it's a malformed CloudFormation template.
+        """
+        policy_documents = []
+        self.obtain_policy_documents(
+            policy_documents=policy_documents, properties=list(self.Properties.__dict__.values())
+        )
+        return policy_documents
+
+    def obtain_policy_documents(self, policy_documents: List[OptionallyNamedPolicyDocument], properties: List[Any]):
+        """
+        Obtains recursively all the optionally named policy documents within a given list of properties.
+        """
+        for property_type in properties:
+            if isinstance(property_type, PolicyDocument):
+                policy_documents.append(OptionallyNamedPolicyDocument(policy_document=property_type, name=None))
+            elif isinstance(property_type, OptionallyNamedPolicyDocument):
+                policy_documents.append(property_type)
+            elif isinstance(property_type, list):
```

Comment:
You will need to handle the Policy type as well.
```
OptionallyNamedPolicyDocument(name=policy.PolicyName, policy_document=policy.PolicyDocument)
```

Reply:
thanks! done so in [50f7936](https://github.com/Skyscanner/pycfmodel/pull/86/commits/50f7936d9db79c3c22e82c4aad50aad270afe505)